server/src/executor/mod.rs

```diff
@@ -130,10 +141,45 @@ impl Executor {
         }
     }
 
+    fn handle_respawn(
```

Comment:
why defining a separate `handle_respawn` function? I think this can be merged into the command handler's `handle` function (do the checks there).

Reply:
Forgot to answer this question, similar reason with the conversation below: Since respawn is using a different cd mechanism, so we need to need to use handle_respawn to check when the respawn command is called, whether it's time to actually handle the respawn command or wait until the cd is done. Maybe the naming here is not good, let me change the name for better understanding

Reply:
I guess it's kind of minor but my thought was why not put the different cd mechanism and all the checks to RespawnHandler::handle? It is equivalent but it helps separating the concerns and hide all the impl details in the command handlers.

Reply:
Yeah, that totally makes sense since first of all I didn't think that was possible. I just finish doing that, and it works well so far